### PR TITLE
ISSUE-690: Allow users to disble server side caching

### LINKED ISSUE #690: Allow users to disable server side caching

### PR DESCRIPTION
**Pull request by raju-saravanan**
Fixes #690

### CODE REVIEW

File: schema-registry/core/src/main/java/com/hortonworks/registries/schemaregistry/DefaultSchemaRegistry.java [hunk 1]
```diff
@@ -999,16 +1001,26 @@ public SchemaBranch getSchemaBranch (Long id) throws SchemaBranchNotFoundExcepti
...
             return value != null ? value : defaultValue;
         }
 
+        private Object getPropertyValue(String propertyKey, Object defaultValue) {
+            Map<String, ?> properties = (Map<String, ?>) getValue(config, "properties", new HashMap<>());
+            return getValue(properties, propertyKey, defaultValue);
+        }
+
+        public Boolean isCacheEnabled() {
+            return (Boolean) getValue(config, ENABLE_CACHING, Boolean.FALSE);
```
Comment (akatona84) [inline schema-registry/core/src/main/java/com/hortonworks/registries/schemaregistry/DefaultSchemaRegistry.java]: It should be enabled by default, to be backward compatible.

**Comment by raju-saravanan** [comment]
@akatona84: Addressed your review comments.

**Comment by guruchai** [comment]
I had to dig around a bit to understand what exactly you are disabling. I am slightly confused with the changes. 
* The code to invalidate the cache in DefaultSchemaRegistry is already commented out. I dont see any calls inflow into HAServerNotificationManager.notify*. It looks the caching is already disabled.

What's the purpose of this change? Is it only avoid to db calls to fetch other instances info?

**Comment by raju-saravanan** [comment]
@guruchai : Caching was not disabled in MASTER. We raised a PR to disable it in CSP-3.0.0, with this change server side cache disabling is configuration from yaml.

**Comment by guruchai** [comment]
lgtm..

**Pull request merged**